tasks/dist.js

```diff
@@ -0,0 +1,106 @@
+module.exports = function(grunt) {
+
+    // location of our temporary working copy we use to build tags
+    var tempCheckout = 'temp/checkout';
+
+    // register a few promise related convenience functions
+    var exec = require('./utils')(grunt).exec;
+
+    var Q = require("q");
+    var fs = require('fs');
+    var simpleGit = require("simple-git");
+
+    // rocky.js filename for a given version
+    function rockyNameForTag(version) {
+        return 'build/dist/rocky-' + version + '.js';
+    }
+
+    // true, if a versioned rocky file exists in the destination folder
+    function versionedRockyExists(tag) {
+        try {
+            return fs.statSync(rockyNameForTag(tag)).isFile();
+        } catch (err) {
+            return false;
+        }
+    }
+
+    // creates a promise that expresses all steps for a given tag
+    function checkoutNpmBuildCopy(tag) {
+        return simpleGit(tempCheckout).promising('checkout', tag)
+        .then(function() {
+            return exec('npm install', {cwd: tempCheckout});
+        })
+        .then(function() {
+            return exec('grunt build', {cwd:tempCheckout});
+        })
+        .then(function() {
+            var taggedPkg = grunt.file.readJSON(tempCheckout + '/package.json');
+            var destName = rockyNameForTag(taggedPkg.version);
+            if (versionedRockyExists(taggedPkg.version)) {
+                grunt.log.warn("skipping file", destName, "Tag", tag, "seems to be named incorrectly.");
+                return false;
+            }
+            return exec('cp ' + tempCheckout + '/' + destName + ' ' + destName);
```

Comment:
These tags are following the aforementioned pattern and should be secure. Will add a comment about it, let me know if you see any other problem.